x-pack/plugin/sql/sql-proto/src/test/java/org/elasticsearch/xpack/sql/proto/SqlVersionTests.java

```diff
@@ -73,4 +75,34 @@ public void testVersionsAndInvalidStringNotEqual() {
         assertNotEquals(ver1, ver2);
     }
 
+    public void testVersionCompatibilityClientWithNoCompatibility() {
+        SqlVersion server = SqlVersion.fromId(CURRENT.id);
```

Comment:
Maybe it's better to use a concrete version here, or a relative major below, to be easy to backport to 7.x without changes.

Reply:
Porting to 7.x shouldn't be an issue, no matter the CURRENT (now being past 7.7.0). The important thing is for the client version to be before 7.7.0 (i.e. lacking the compatibility capability, adding its version to requests etc.).